### PR TITLE
Add pkg-config file

### LINKED ISSUE #585: Provide pkg-config file for Sandia OpenSHMEM
### Feature Request

Provide `pkg-config` file(s) for Sandia OpenSHMEM.

### PR DESCRIPTION
**Pull request by jdinan**
Closes #585

### CODE REVIEW

File: src/sos.pc.in [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: sos
+Description: @PACKAGE_NAME@
+URL: @PACKAGE_URL@
+Version: @VERSION@
+Libs: -L${libdir} @WRAPPER_COMPILER_EXTRA_LDFLAGS@ -lsma @WRAPPER_COMPILER_EXTRA_LIBS@
```
Comment (davidozog) [inline src/sos.pc.in]: Interesting: no need for `LDFLAGS` here?  And `CFLAGS` below?

File: src/sos.pc.in [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+Cflags: -I${includedir}
```
Comment (davidozog) [inline src/sos.pc.in]: Might we also want the `${includedir}/mpp` here?  I wouldn't think so, but just making sure.

File: src/sos.pc.in (hunk 1 above)
Comment (jdinan) [reply src/sos.pc.in]: Nope, pkg-config is a little underwhelming like that.

File: src/sos.pc.in (hunk 2 above)
Comment (jdinan) [reply src/sos.pc.in]: Nope, that would be redundant, since the same headers are already installed in `${includedir}`.

Comment (jdinan) [reply src/sos.pc.in]: BTW, `pkg-config --cflags` is identical to `CFLAGS` and `pkg-config --libs` is identical to `LDFLAGS` generated by the `oshcc` wrapper.  So, if we ever find an issue here, both places need an update.

**Pull request merged**